File: auth-service/src/auth/auth.controller.ts
import { Controller, Post, Body } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { AuthService } from './auth.service';
import { LoginDto, RecoverDto, ResetDto } from './dto/auth.dto';

@ApiTags('auth')
@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) { }

  @Post('login')
  @ApiOperation({ summary: 'Login and get Access JWT' })
  login(@Body() loginDto: LoginDto) {
    return this.authService.login(loginDto);
  }

  @Post('recover-password')
  @ApiOperation({ summary: 'Initiate password recovery flow' })
  recoverPassword(@Body() recoverDto: RecoverDto) {
    return this.authService.recoverPassword(recoverDto.email);
  }

  @Post('reset-password')
  @ApiOperation({ summary: 'Set a new password using a reset token' })
  resetPassword(@Body() resetDto: ResetDto) {
    return this.authService.resetPassword(resetDto);
  }
}

File: auth-service/src/main/java/com/empresa/auth_service/controller/AuthController.java
package com.empresa.auth_service.controller;

import com.empresa.auth_service.dto.LoginDto;
import com.empresa.auth_service.dto.RecoverDto;
import com.empresa.auth_service.dto.ResetDto;
import com.empresa.auth_service.service.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.Optional;
import com.empresa.auth_service.model.User;
import com.empresa.auth_service.repository.UserRepository;
import com.empresa.auth_service.security.JwtUtil;

@RestController
@RequestMapping("/auth")
public class AuthController {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private JwtUtil jwtUtil;

    @Autowired
    private AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginDto loginDto) {
        try {
            return ResponseEntity.ok(authService.login(loginDto));
        } catch (RuntimeException e) {
            return ResponseEntity.status(401).body(Map.of("message", e.getMessage()));
        }
    }

    @PostMapping("/recover-password")
    public ResponseEntity<?> recoverPassword(@RequestBody RecoverDto recoverDto) {
        try {
            return ResponseEntity.ok(authService.recoverPassword(recoverDto.getEmail()));
        } catch (RuntimeException e) {
            return ResponseEntity.status(404).body(Map.of("message", e.getMessage()));
        }
    }

    @PostMapping("/reset-password")
    public ResponseEntity<?> resetPassword(@RequestBody ResetDto resetDto) {
        try {
            return ResponseEntity.ok(authService.resetPassword(resetDto));
        } catch (RuntimeException e) {
            return ResponseEntity.status(400).body(Map.of("message", e.getMessage()));
        }
    }

    @GetMapping("/health")
    public ResponseEntity<?> health() {
        return ResponseEntity.ok(Map.of(
            "status", "UP",
            "service", "auth-service"
        ));
    }
}
